### PR TITLE
Session manager

### PR DESCRIPTION
**Pull request by ank42**

### CODE REVIEW

File: l0/terraform.tfstate [hunk 1]
```diff
@@ -14,7 +14,7 @@
         {
           "schema_version": 1,
           "attributes": {
-            "arn": "arn:aws:dynamodb:us-west-2:175817389458:table/for_state_lock",
+            "arn": "arn:aws:dynamodb:us-west-2:397343975353:table/for_state_lock",
             "attribute": [
               {
                 "name": "LockID",
```
Comment (vladimirmukhin) [inline l0/terraform.tfstate:20]: Not sure why this file is constantly being changed in your pull requests.
Are you making and adjustments in level0?

File: l0/terraform.tfstate.backup [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            "website_domain": null,
+            "website_endpoint": null
+          },
+          "sensitive_attributes": [],
+          "private": "bnVsbA=="
+        }
+      ]
+    }
+  ]
+}
```
Comment (vladimirmukhin) [inline l0/terraform.tfstate.backup]: Please remove `terraform.tfstate.backup` it's enough that we just have `terraform.tfstate` committed in git

File: modules/asg/main.tf [hunk 3]
```diff
@@ -20,12 +20,7 @@ resource "aws_security_group" "asg_sg" {
     cidr_blocks = ["0.0.0.0/0"]
   }
 
-  ingress {
-    from_port   = 22
-    to_port     = 22
-    protocol    = "tcp"
-    cidr_blocks = ["10.0.0.0/16"]
-  }
+# removed port 22
```
Comment (vladimirmukhin) [inline modules/asg/main.tf]: no need to have this comment

```suggestion
```

File: modules/asg/main.tf [hunk 4]
```diff
@@ -50,13 +45,12 @@ resource "aws_security_group" "asg_sg" {
 resource "aws_launch_template" "t3micro" {
   name_prefix            = "t3micro"
   image_id               = data.aws_ami.amazon-linux-2.id
-  instance_type          = "t3.micro"
-  key_name               = "MyKeyPair"
+  instance_type          = "t3.micro"                       #Removed key variable
```
Comment (vladimirmukhin) [inline modules/asg/main.tf]: No need to have this comment
```suggestion
  instance_type          = "t3.micro"
```

File: modules/ec2/main.tf [hunk 5]
```diff
@@ -13,7 +13,7 @@ resource "aws_instance" "Dev" {
   instance_type               = "t3.micro"
   subnet_id                   = var.subnet_ids[0]
   security_groups             = [aws_security_group.main.id]
-  key_name                    = "MyKeyPair"
+  #key_name                    = "MyKeyPair"
```
Comment (vladimirmukhin) [inline modules/ec2/main.tf]: Having commented-out code is a bad practice. When you remove something, just remove it.
Don't worry about having to revert things - the history will be still tracked in git
```suggestion
```

File: modules/vpc/main.tf [hunk 6]
```diff
@@ -30,10 +30,7 @@ resource "aws_subnet" "Private" {
     Name = "private${count.index}"
   }
 }
-output "private_subnet_id" {
-  description = "Private Subnet ID"
-  value       = aws_subnet.Private.*.id
-}
```
Comment (vladimirmukhin) [inline modules/vpc/main.tf:36]: Please don't mix changes not related to the feature in a single PR

File: test/.terraform.lock.hcl [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    "zh:730284250fd949a59afb6935b3a68a33709d5a78b686fa98f351ad32c919cfc3",
+    "zh:7461ebd6fb35900d620cfa3f42126d988ea1e604ee3828d1c64d5727f908bd26",
+    "zh:7c85743b31c7459f8e74aaa98471ba82c54517eb908603411808a12982d89b1c",
+    "zh:8ed977b7fb97de624f5414b08cab36fd973a624072e0e9082c0c822e0864c7b9",
+    "zh:94ae7313bb0b425d4007a0b70601a337972c4f0f7a323487acf69215e74b4425",
+    "zh:b5a1589672d709da725a72c46d28bf5b2dea71325f6e0b44a0049f644cd09eba",
+    "zh:c7e8e7ce59e4578416557fc2f138137af3c8365ac3e34f0ff5166323c7d641a1",
+    "zh:ccf2e286b207e749fff76bb4075deddb9e7e237936d8654f34828c54e7035455",
+  ]
+}
```
Comment (vladimirmukhin) [inline test/.terraform.lock.hcl]: Please remove the folder test

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: l0/terraform.tfstate (hunk 1 above)
Comment (ank42) [reply l0/terraform.tfstate:20]: No i am Just applying it!

File: l0/terraform.tfstate.backup (hunk 2 above)
Comment (ank42) [reply l0/terraform.tfstate.backup]: Resolved

File: modules/asg/main.tf (hunk 3 above)
Comment (ank42) [reply modules/asg/main.tf]: resolved

File: modules/asg/main.tf (hunk 4 above)
Comment (ank42) [reply modules/asg/main.tf]: resolved

File: modules/ec2/main.tf (hunk 5 above)
Comment (ank42) [reply modules/ec2/main.tf]: resolved

File: modules/vpc/main.tf (hunk 6 above)
Comment (ank42) [reply modules/vpc/main.tf:36]: resolved!

File: test/.terraform.lock.hcl (hunk 7 above)
Comment (ank42) [reply test/.terraform.lock.hcl]: Resolved

File: modules/vpc/main.tf (hunk 6 above)
Comment (vladimirmukhin) [reply modules/vpc/main.tf:36]: Are you sure, I still see this change from my side

File: l0/terraform.tfstate.backup (hunk 2 above)
Comment (vladimirmukhin) [reply l0/terraform.tfstate.backup]: The file is still there. Please check

File: modules/asg/main.tf (hunk 3 above)
Comment (vladimirmukhin) [reply modules/asg/main.tf]: Still not fixed

File: modules/asg/main.tf (hunk 4 above)
Comment (vladimirmukhin) [reply modules/asg/main.tf]: Still not fixed

File: modules/ec2/main.tf (hunk 5 above)
Comment (vladimirmukhin) [reply modules/ec2/main.tf]: Still not fixed

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**